bot/src/com/mygdx/game/bot/screen/GameScreen.java

```diff
@@ -133,20 +181,76 @@ private void runBotIteration() {
     }
     log.info("start turn");
     var unit = nextUnitUtil.selectNextUnit();
-    if (unit == 0xC0FFEE) { // todo ensure the case where a unit that has moveRange and attack but no possible
-      // ways to use it be skipped
-      log.info("end turn");
-      endTurn();
+    if (unit == 0xC0FFEE) {
+      buildBuildingsPrepare();
       return;
     }
     var field = nextFieldUtil.selectFieldInRangeOfUnit(unit);
     log.info("moving entity");
+    if (field == 0xC0FFEE) {
+      buildBuildingsPrepare();
+      return;
+    }
     moveEntityService.moveEntity(unit, coordinatesComponentMapper.get(field));
     botAttackUtil.attack(unit);
   }
```

Comment:
(jak przeczytasz wiadomość na discordzie)

to w tym miejscu trzeba jedynie się decydować czy my chcemy atakować czy chodzić, losowo może 

szczerze mówiąc to nadal powinien być błąd z tym, że dwa razy to będzie się wykonywało, bo changesApplied będzie dwa razy wywoływane, ale jak ci działa to idk

Reply:
No narazie zostawimy jak jest, move -> potem attack, w innym prze można to poprawić